src/components/app.tsx

```diff
@@ -10,8 +10,43 @@ import {EncodingPane} from './encoding-pane';
 import {Header} from './header';
 import {ViewPane} from './view-pane';
 
+import {
+  ActionHandler,
+  DatasetAsyncAction,
+  datasetUrlLoad,
+} from '../actions';
+
+
+
+interface DataURL {
+  name: string;
+  url: string;
+};
+export type VoyagerData = DataURL | Object[];
```

Comment:
@kanitw this has been done. with one caveat. it seems that UrlData or InlineData do not have names. For now i've just hard coded "custom data" as a name. I can get your thoughts in another thread of why NamedData is separate and whether the other two data types could have optional names. (Or whether i should just extend one with the other—though i feel like that would be a new custom type, which i am trying to avoid if possible).